### ISSUE TITLE
ActiveRecord::PendingMigrationError is not raised when non-primary database migration is down with multi-db

### ISSUE THREAD

**Issue by suketa** [rails/rails#37524]
### Steps to reproduce
ActiveRecord::PendingMigrationError is not raised when non-primary database migrations are down with multi-db.

config database.yml 

```
development:
  backbone:
    <<: *default
    database: backbone_development
  library:
    <<: *default
    database: library_development
    migrations_paths: db/library_migrate
```

run the following
```
bin/rails g scaffold User title 
bin/rails g scaffold Book title --db library
bin/rails db:create db:migrate
bin/rails s
# in another console
bin/rails g migration add_published_at_to_books published_at:datetime --db library
```
access http://localhost:3000/users or http://localhost:3000/books

### Expected behavior
Raise ActiveRecord::PendingMigrationError because add_published_at_to_books migration status is down

```
$ bin/rails db:migrate:status

database: backbone_development
 Status   Migration ID    Migration Name
--------------------------------------------------
   up     20191022014341  Create users

database: library_development

 Status   Migration ID    Migration Name
--------------------------------------------------
   up     20191022023152  Create books
  down    20191022023755  Add published at to books
```

```
$ rails db:abort_if_pending_migrations
You have 1 pending migration:
  20191022023755 AddPublishedAtToBooks
Run `rails db:migrate` to update your database then try again.
```
### Actual behavior
There is no error

### System configuration
Rails 6.0.0
Ruby 2.6.5

**Comment by eileencodes** [comment]
I was able to reproduce this but it's non-trivial to fix. While we could check other connections there's still an underlying issue. When we boot the server in development mode the app is set to not eager load which means that even if we were looping through all the connections the point in which we check for pending migrations those other models won't be connected. I'm not sure how to fix this yet but wanted to add a note that I looked at it.
